src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -92,6 +243,10 @@ public DiskCache(string requestPath, string fullPath, string querystring)
         /// </returns>
         public override async Task<bool> IsNewOrUpdatedAsync()
         {
+            //TODO: Before this check is performed it should be throttled. For example, only perform this check 
```

Comment:
I have code in a few projects that we do a very similar thing. We do this in Umbraco core and in Examine too. Using a thread timer we start the timer whenever this is called, there's 2 values that need to be checked: timer timeout (i.e. 5 seconds) and the max timeout (i.e. 30 seconds). This ends up being a sliding offset, the logic won't be executed if this method keeps getting executed until the timer reaches the max (30 seconds) or if the last request time is larger than 5 seconds.

This means that under high load the perf of this would be much better since it wouldn't have to read from the file system on every request just to check if its out of date. Of course there would be a max of 30 seconds under load if the file did change but that is OK.

I can make another PR for this when I get time - I'll create an issue for it.